controllers/component_controller_test.go

```diff
@@ -783,12 +789,18 @@ var _ = Describe("Component controller", func() {
 			createdHasComp := &appstudiov1alpha1.Component{}
 			Eventually(func() bool {
 				k8sClient.Get(context.Background(), hasCompLookupKey, createdHasComp)
-				return len(createdHasComp.Status.Conditions) > 0 && createdHasComp.Status.Conditions[len(createdHasComp.Status.Conditions)-1].Status == metav1.ConditionFalse
+				return len(createdHasComp.Status.Conditions) > 1 && createdHasComp.Status.Conditions[len(createdHasComp.Status.Conditions)-1].Status == metav1.ConditionFalse
```

Comment:
should here check `Conditions[len(createdHasComp.Status.Conditions)-2].Status`?

Reply:
Shouldn't be needed. Because we add the `Created` status condition _after_ the gitops status condition, waiting til the `Created` condition is added is sufficient to "verify" that the reconcile has concluded.

We also check `Conditions[len(createdHasComp.Status.Conditions)-2].Status` a few lines down below.